testutil/helper.go

```diff
@@ -188,47 +184,55 @@ func GetUser(name, id string, config Config) *User {
 }
 
 func CheckPet(t *testing.T, db *gorm.DB, pet Pet, expect Pet) {
+	doCheckPet(t, db, pet, expect, false)
+}
+
+func doCheckPet(t *testing.T, db *gorm.DB, pet Pet, expect Pet, unscoped bool) {
 	if pet.ID != 0 {
 		var newPet Pet
 		if err := db.Where("id = ?", pet.ID).First(&newPet).Error; err != nil {
 			t.Fatalf("errors happened when query: %v", err)
 		} else {
-			AssertObjEqual(t, newPet, pet, "ID", "CreatedAt", "UpdatedAt", "UserID", "Name")
-			AssertObjEqual(t, newPet, expect, "ID", "CreatedAt", "UpdatedAt", "UserID", "Name")
+			AssertObjEqual(t, newPet, pet, "ID", "CreatedAt", "UpdatedAt", "DeletedAt", "UserID", "Name")
+			AssertObjEqual(t, newPet, expect, "ID", "CreatedAt", "UpdatedAt", "DeletedAt", "UserID", "Name")
 		}
 	}
 
-	AssertObjEqual(t, pet, expect, "ID", "CreatedAt", "UpdatedAt", "UserID", "Name")
+	AssertObjEqual(t, pet, expect, "ID", "CreatedAt", "UpdatedAt", "DeletedAt", "UserID", "Name")
```

Comment:
Can we use `reflect.DeepEqual` to compare structs directly, instead of comparing each fields seperately?